next/components/forms/FormPageWrapper.tsx

```diff
@@ -28,48 +28,42 @@ export type FormPageWrapperProps = {
   ssrCurrentAuthProps?: GetSSRCurrentAuth
 }
 
-// keeps hash, removes trailing id along with last slash: /mestske-sluzby/slug/id#hash -> /mestske-sluzby/slug#hash
-const matchUrlWithoutId = /(\/mestske-sluzby\/[^/]+)\/[^#/]+/
-
 // custom plausible tracking - we exclude '/mestske-sluzby/*/*' in top level plausible provider
-// instead, we track the url without the UUID
-// this is done on '/mestske-sluzby/slug#hash' url, note that '/mestske-sluzby/slug/whatever#hash' would still be ignored even with manual tracking
-// reference https://plausible.io/docs/custom-locations
-const useCustomPlausibleFormPagesTracking = () => {
+// instead, we track these as custom events: formSlug#hash
+const useCustomPlausibleFormPagesTracking = (formSlug: string) => {
   const router = useRouter()
   const plausible = usePlausible()
   const firstRender = useRef(true)
 
   useEffect(() => {
     const onHashChangeStart = (url: string) => {
-      const urlWithoutId = url?.replace(matchUrlWithoutId, '$1')
-      plausible('pageview', { u: urlWithoutId })
+      const hash = url.split('#')[1]
+      plausible(`${formSlug}#${hash}`)
     }
 
     router.events.on('hashChangeStart', onHashChangeStart)
 
     return () => {
       router.events.off('hashChangeStart', onHashChangeStart)
     }
-  }, [plausible, router.events])
+  }, [formSlug, plausible, router.events])
 
   // track initial pageview
   useEffect(() => {
```

Comment:
useEffectOnce?

Reply:
true, have forgotten that exists. When I go around this next time I'll replace

Reply:
oh, this isn't merged yet 😅 nevermind, fixing now